docs/src/main/asciidoc/elasticsearch.adoc

```diff
@@ -0,0 +1,392 @@
+
+1. We inject an Elasticsearch low level `RestClient` inside the service.
+2. We create an Elasticsearch request.
+3. We use Vert.X `JsonObject` to serialize the object before sending it to Elasticsearch, you can use whatever you want to serialize to JSON.
+4. We send the request (indexing request here) to Elasticsearch.
+5. In order to deserialize the object from Elasticsearch, we again use Vert.X `JsonObject`.
+
+Now, edit the `org.acme.elasticsearch.FruitResource` class as follows:
+
+[source,java]
+----
+@Path("/fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class FruitResource {
+    @Inject
+    FruitService fruitService;
+
+    @POST
+    public Response index(Fruit fruit) throws IOException {
+        if (fruit.id == null) {
+            fruit.id = UUID.randomUUID().toString();
+        }
+        fruitService.index(fruit);
+        return Response.created(URI.create("/fruits/" + fruit.id)).build();
+    }
+
+    @GET
+    @Path("/{id}")
+    public Fruit get(@PathParam("id") String id) throws IOException {
+        return fruitService.get(id);
+    }
+
+    @GET
+    @Path("/search")
+    public List<Fruit> search(@QueryParam("name") String name, @QueryParam("color") String color) throws IOException {
+        if (name != null) {
+            return fruitService.searchByName(name);
+        } else if (color != null) {
+            return fruitService.searchByColor(color);
+        } else {
+            throw new BadRequestException("Should provide name or color query parameter");
+        }
+    }
+
+}
+----
+
+The implementation is pretty straightforward and you just need to define your endpoints using the JAX-RS annotations and use the `FruitService` to list/add new fruits.
+
+== Configuring Elasticsearch
+The main property to configure is the URL to access to Elasticsearch.
+
+A sample configuration should look like this:
+
+[source,properties]
+----
+# configure the Elasticsearch client to a cluster of two nodes
+quarkus.elasticsearch.hosts = elasticsearch1:9200,elasticsearch2:9200
+----
+
+In this example, we are using a single instance running on localhost:
+
+[source,properties]
+----
+# configure the mongoDB client for a single instance on localhost
+quarkus.elasticsearch.hosts = localhost:9200
+----
+
+If you need more configuration properties, there is a full list at the end of this guide.
+
+
+== Running an Elasticsearch
+As by default, the Elasticsearch client is configured to access a local Elasticsearch on port 9200 (the default Elasticsearch port),
+if you have a local running database on this port, there is nothing more to do before being able to test it!
+
+If you want to use Docker to run an Elasticsearch database, you can use the following command to launch one:
+[source,shell]
+----
+docker run --name elasticsearch  -e "discovery.type=single-node" -e "ES_JAVA_OPTS=-Xms512m -Xmx512m"\
+       --rm -p 9200:9200 docker.elastic.co/elasticsearch/elasticsearch-oss:7.6.0
```

Comment:
I check other guides and we always hardcode the mongo/kafka/neo4j version inside the guide (or as attribute of the guide not coming from the bom).
I assume this is easier and allow using the same inside the integration test ...

Reply:
This is because the properties are not inherited from the bom so they are not available.

It's a bit different in the Elasticsearch case because we already have `elasticsearch-server.version` in the `build-parent/pom.xml` and this property is available and can be pushed as an Asciidoc attribute.

Reply:
Done.
I also update the hibernate search elasticsearch guide to use the same version from the build-parent.